.github/workflows/net-utils-builder-staging.yml

```
@@ -0,0 +1,63 @@
+name: Build and push net utility staging image
+
+on:
+  pull_request_target:
+    branches:
+      - 'main'
+    paths:
+      - 'containers/utility/Dockerfile'
+
+jobs:
+  build-and-push-staging:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+
+    # safe to use pull_request_target because we only run this for trusted collaborators
+    if: contains(fromJson('["OWNER","MEMBER","COLLABORATOR"]'), github.event.pull_request.author_association)
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
```

I think you need to add this part so it will checkout the PR and not main

```suggestion
  with:
    ref: ${{ github.event.pull_request.head.sha }}
```

---

thanks, added (and updated release to `v5`)